TheFunctionChangedMyMind/README-en.md

```diff
@@ -0,0 +1,322 @@
+---
+title: The Function that changed my mind
+published: true
+description: How a single function showed me a lot of Kotlin
+tags: kotlin,functions,android,refactor
+cover_image:
+series:
+---
+
+When writting code, we should always have a place for improvement. Improvement can present itself in different ways. A proof of it is how in recent years the attention to CI (_Continuous Integration_) and CD (_Continuous Delivery_) has grown significantly. Same goes for new languages and technologies.
+
+But not all involve changing to a new language, technology or tool. Some of these improvements are simply changing how something works, adding tests, adding abstractions, changing patterns, etc. This is the case for the change I introduced on one of my projects.
```

Comment:
Nevertheless, this doesn't involve changing to a new language, technology, nor tool. Some of these improvements are so simple as changing how something works, i. e. adding tests/abstractions, changing patterns, etc. This is how I have proceeded in several projects I have worked in.